search_service/proxy/gcp_data_catalog.py

```diff
@@ -0,0 +1,248 @@
+# Copyright Contributors to the Amundsen project.
+# SPDX-License-Identifier: Apache-2.0
+
+import logging
+import re
+
+from google.api_core.client_options import ClientOptions
+from google.cloud import datacatalog_v1
+from typing import Any, List, Dict, Union, Optional
+
+from search_service.models.dashboard import SearchDashboardResult, Dashboard
+from search_service.models.table import SearchTableResult
+from search_service.models.table import Table
+from search_service.models.user import SearchUserResult
+from search_service.proxy import BaseProxy
+
+logging.basicConfig(level=logging.DEBUG)
+
+LOGGER = logging.getLogger(__name__)
+
+
+# @todo - consider moving base class with init to commons - it's exactly the same in metadatalibrary
+class GCPDataCatalogProxy(BaseProxy):
+    def __init__(self, *,
+                 project_id: str = '',
+                 credentials_file: Optional[str] = None,
+                 client: datacatalog_v1.DataCatalogClient = None,
+                 page_size: int = 10,
+                 **kwargs
+                 ) -> None:
+
+        # Scope object will limit all our Data Catalog searches to configured project_id
+        self.scope = datacatalog_v1.types.SearchCatalogRequest.Scope()
+        self.scope.include_project_ids.append(project_id)
+
+        client_options = ClientOptions(scopes=[self.scope])
+
+        if credentials_file:
+            _client = datacatalog_v1.DataCatalogClient.from_service_account_file(credentials_file,
+                                                                                 client_options=client_options)
+        else:
+            _client = datacatalog_v1.DataCatalogClient(credentials=credentials_file, client_options=client_options)
```

Comment:
I think datacatalog_v1.DataCatalogClient(credentials=None, client_options=client_options) or 
datacatalog_v1.DataCatalogClient(client_options=client_options) would be better. With credentials_file one might wander if it shouldn't be None